src/node-save-pixels.ts

```diff
@@ -1,13 +1,23 @@
 import { NdArray } from 'ndarray';
-import sharp, { AvailableFormatInfo } from 'sharp';
+import sharp, { FormatEnum } from 'sharp';
 import { putPixelData } from './common';
 
 export async function savePixelsInternal(
 	pixels: NdArray<Uint8Array | Uint8ClampedArray>,
-	mimeType: string
+	mimeType: string,
+	quality?: number
 ): Promise<Uint8Array> {
 	const [width, height, channels] = pixels.shape as [number, number, 4];
 	const data = putPixelData(pixels, new Uint8Array(width * height * channels));
-	const format = mimeType.replace('image/', '') as unknown as AvailableFormatInfo;
-	return sharp(data, { raw: { width, height, channels } }).toFormat(format).toBuffer();
+	const format = mimeType.replace('image/', '') as keyof FormatEnum;
+
+	const options = {
+		// Applicable to most formats.
+		// Where used, an integer between 1 and 100
+		quality: quality == undefined ? undefined : Math.round(1 + quality * 99),
+		// applicable to some formats, notably webp, avif
+		lossless: quality === 1,
+	};
```

Comment:
I'm probably being paranoid here, but could you split this out so that the `quality` parameter to sharp is only assigned if there's a `quality` input? Something like:

```js
const options = {};

if (typeof quality === 'number') {
  options.quality = Math.round(1 + quality * 99);
  options.lossless = quality === 1;
}
```

I've had issues in the past where passing Sharp's default quality value does something different than omitting it (enables palette compression for PNG) and I just want to be sure there's no similar effect for `undefined`.

Reply:
I'm probably being paranoid here, but could you split this out so that the `quality` parameter to sharp is only assigned if there's a `quality` input? Something like:

You are being paranoid, but it's warranted due to the semantics `sharp` applies to the `quality` option.

Here's the underlying logic in the library: https://github.com/lovell/sharp/blob/da655a1859744deec9f558effa5c9981ef5fd6d3/lib/output.js#L571-L583

Even if `quality` is passed, I think it's a bad idea to use it. Maybe I should set `palette: false` instead?

Reply:
Assuming `palette` just affects PNG, that seems like a good thing to disable, yes! Or we could avoid passing `quality` for PNG textures, either is fine with me.